cartoframes/data/observatory/repository/dataset_repo.py

```diff
@@ -20,7 +20,9 @@ def __init__(self):
 
     def get_all(self, filters=None, credentials=None):
         self.client.set_user_credentials(credentials)
-        return self._get_filtered_entities(filters)
+        response = self._get_filtered_entities(filters)
+        self.client.set_user_credentials(None)
```

Comment:
It would be nice having a new method for that, instead of doing a "set" and "unset". Also, it allows us to avoid adding specific logic for this case in `get_geographies` and `get_datasets` in the repo

Reply:
Yes. We should revisit this part